src/model/event.ts

```diff
@@ -30,11 +31,42 @@ export const isPastEvent = (event: IEvent): boolean =>
 export const isComingEvent = (event: IEvent): boolean =>
   DateTime.fromISO(event.date) >= DateTime.now();
 
-export const filterPastEvents = (events: IEvent[]) =>
-  events.filter(isPastEvent);
+export const isPublished = (event: IEvent): boolean =>
+  (event.published ?? 'true') == 'true';
 
-export const filterComingEvents = (events: IEvent[]) =>
-  events.filter(isComingEvent);
+export const filterPastEvents = (
```

Comment:
From my personal point of view, functions should be as simple as possible and they should solve one specific task (Single Responsibility principle and Keep cognitive complexity low)

I guess we have all the pieces to get published or unpublished events, we have only to combine them.

```typescript
// i.e if we want only published past events
const allpastEvents = filterPastEvents(events);
const pastPublishedEvents = pastEvents.filter(isPublished);
const pashUnpublishedEvents = pastEvents.filter(pe => !isPublished(pe));
```

This is my personal taste! so feel free to skip this suggestion. I prefer multiple simple pieces and then combine them, instead of empowering functions